wagtail_streamfield_migration_toolkit/management/commands/streamdatamigration.py

```diff
@@ -0,0 +1,230 @@
+from functools import lru_cache
+from django.core.management.base import BaseCommand, CommandError
+from django.db import connection
+from django.db.migrations.loader import MigrationLoader
+from django.db.migrations import Migration
+from django.db.migrations.autodetector import MigrationAutodetector
+from django.db.migrations.state import ProjectState
+from django.db.migrations.writer import MigrationWriter
+from django.apps import apps
+from wagtail.blocks import StreamBlock, StructBlock
+
+from wagtail_streamfield_migration_toolkit.migrate_operation import MigrateStreamData
+from wagtail_streamfield_migration_toolkit.operations import (
+    RemoveStreamChildrenOperation,
+    RenameStreamChildrenOperation,
+    RemoveStructChildrenOperation,
+    RenameStructChildrenOperation,
+)
+
+
+class Command(BaseCommand):
+
+    help = "Generate data migrations with the given operations. Note that you can only \
+        generate changes for models in a single app at a time."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--name", help="Use this name for migration file")
+        subparsers = parser.add_subparsers(dest="operation_type")
+
+        rename_parser = subparsers.add_parser("rename", help="rename --help")
+        rename_parser.add_argument("app_label", help="app name")
+        rename_parser.add_argument("old_name", help="name of the block to be renamed")
+        rename_parser.add_argument("new_name", help="new name")
+        rename_parser.add_argument(
```

Comment:
Similarly, could arguments like paths, block_name, etc be on the main parser since they don't differ between the subparsers? Or does that make the help text confusing?

Reply:
The block name kind of differs. Like it's not exactly supposed to be part of the path, but an argument used when renaming. If we add alter value for example it'll take a new value instead of a block name argument.

Reply:
Ah, I see, this is because of alter value then? I see what you mean, but even there, it might be a bit weird for users to have to specify a path to a parent block for rename, and a path to the child block for alter value - feels a bit unintuitive. Maybe we should revisit how we define block path in the future.. Still, this is fine as is for now - let's leave it.